lib/datasource/github-releases/index.ts

```diff
@@ -66,11 +74,183 @@ export async function getReleases({
+    for (const line of res.body.split('\n')) {
+      const [lineDigest, lineFn] = line.split(/\s+/, 2);
+      if (lineFn === releaseFilename) {
+        return lineDigest;
+      }
+    }
+  } else {
+    const res = http.stream(releaseAsset.browser_download_url);
+    const algorithm = inferHashAlg(digestAsset.currentDigest);
+    const newDigest = await hasha.fromStream(res, { algorithm });
+    return newDigest;
+  }
+  logger.debug({ releaseAsset }, 'fetch');
+  return null;
+}
+
+export async function getGithubRelease(
+  apiBaseUrl: string,
+  repo: string,
+  version: string
+): Promise<GithubRelease> {
+  const url = `${apiBaseUrl}repos/${repo}/releases/tags/${version}`;
+  const res = await http.getJson<GithubRelease>(url);
+  return res.body;
+}
+
+function getDigestCacheKey(
+  { lookupName: repo, currentValue, currentDigest, registryUrl }: DigestConfig,
+  newValue: string
+): string {
+  const type = 'digest';
+  return `${registryUrl}:${repo}:${currentValue}:${currentDigest}:${newValue}:${type}`;
+}
+
+/**
+ * github.getDigest
+ *
+ * The `newValue` supplied here should be a valid tag for the GitHub release.
+ * Requires `currentValue` and `currentDigest`.
+ *
+ * There may be many assets attached to the release. This function will:
+ *  - Identify the asset pinned by `currentDigest` in the `currentValue` release
+ *     - Download small release assets, parse as checksum manifests (e.g. `SHASUMS.txt`).
+ *     - Download individual assets until `currentDigest` is encountered. This is limited to sha256 and sha512.
+ *  - Map the hashed asset to `newValue` and return the updated digest as a string
+ */
+export async function getDigest(
+  { lookupName: repo, currentValue, currentDigest, registryUrl }: DigestConfig,
+  newValue?: string
+): Promise<string | null> {
+  logger.debug(
+    { repo, currentValue, currentDigest, registryUrl, newValue },
+    'getDigest'
+  );
+  if (!currentDigest) {
+    return null;
+  }
+  if (!currentValue) {
+    return currentDigest;
+  }
+  const cacheKey = getDigestCacheKey(
+    { lookupName: repo, currentValue, currentDigest, registryUrl },
+    newValue
+  );
+  const cachedResult = await packageCache.get<string>(cacheNamespace, cacheKey);
+  // istanbul ignore if
+  if (cachedResult) {
+    return cachedResult;
+  }
+
+  const apiBaseUrl = getApiBaseUrl(getSourceUrlBase(registryUrl));
+  const currentRelease = await getGithubRelease(apiBaseUrl, repo, currentValue);
+  let digestAsset = await findDigestFile(currentRelease, currentDigest);
+  if (!digestAsset) {
+    digestAsset = await findAssetWithDigest(currentRelease, currentDigest);
+  }
+  let newDigest: string;
+  if (!digestAsset || newValue === currentValue) {
+    newDigest = currentDigest;
+  } else {
+    const newRelease = await getGithubRelease(apiBaseUrl, repo, newValue);
+    newDigest = await findNewDigest(digestAsset, newRelease);
+  }
+
+  const cacheMinutes = 10;
```

Comment:
Wouldn't this result change very infrequently? In which case we can have an aggressive cache time

Reply:
That's correct - I've increased the cache duration from 10 minutes to 1 day, please advise if it should be more aggressive than that.